scripts/codestyle/run_uncrustify.sh

```diff
@@ -13,8 +13,12 @@ while getopts :qh opt; do
 	h)
 		echo "run_uncrustify.sh [options] [<git diff args>]"
 		echo ""
-		echo "Check codestyle of a change. Change generated by bypassing"
-		echo "diff-generating arguments to git-diff"
+		echo "Check codestyle of a change. The change is generated by passing"
+		echo "<git diff args> to git diff. Like:"
+		echo ""
+		echo "run_uncrustify.sh                      # check working tree"
+		echo "run_uncrustify.sh --cached             # check index"
+		echo "run_uncrustify.sh origin/master...HEAD # check current branch"
```

Comment:
That's a good question! Usually I try both until I'm satisfied with result :-)
Seems like in diff case triple-dot form is likely desirable.
```
       git diff [--options] <commit> <commit> [--] [<path>...]
           This is to view the changes between two arbitrary <commit>.

       git diff [--options] <commit>..<commit> [--] [<path>...]
           This is synonymous to the previous form. ...
       
       git diff [--options] <commit>...<commit> [--] [<path>...]
           This form is to view the changes on the branch containing and up to the second <commit>, starting at a common ancestor of both <commit>. "git
           diff A...B" is equivalent to "git diff $(git-merge-base A B) B".
```

I assume usually one want to check their branch changes without changes from master. Let's suppose such (we were about to break record of number of messages in a talk about git without a picture, so I'm sorry :-))
```
$ tig
db467e3 Anton Kozlov         M─┐ [master] {origin/master} Merge pull request #1098 from embox/fix-xen-build
fb87b2c Anton Kozlov         │ o [HEAD] [fix-xen-build]  Remove termios dependency on sched
3fe7991 Anton Kozlov         M─│─┐ Merge pull request #1099 from embox/xen-vagrant
79be763 Anton Kozlov         │ │ o Add vagrant defintion for xen dev vm
7184093 Anton Bondarev       M─┴─┘ Merge pull request #1097 from embox/openlibm-improve
...

$ git --no-pager diff --stat 3fe7991..fb87b2c 
 .gitignore                    |  3 ---
 Vagrantfile                   | 27 ---------------------------
 src/drivers/tty/termios_ops.c |  4 +---
 src/drivers/tty/termios_ops.h |  2 ++
 src/drivers/tty/tty.c         |  4 ++++
 5 files changed, 7 insertions(+), 33 deletions(-)

$ git --no-pager diff --stat 3fe7991...fb87b2c 
 src/drivers/tty/termios_ops.c | 4 +---
 src/drivers/tty/termios_ops.h | 2 ++
 src/drivers/tty/tty.c         | 4 ++++
 3 files changed, 7 insertions(+), 3 deletions(-)
```
Note that tripple dot don't try to revert 3fe7991 just because it's not accessible from current branch.

Funny part, that git log takes revset described in gitrevisions(7).Double and tripple dot notation have _opposite_ semantics there. I don't know what forced git developers to make so.